### ISSUE TITLE
h5md_parallel test fails

### ISSUE THREAD

**Issue by junghans** [espressopp/espressopp#426]
```
77/85 Test #76: h5md_parallel ........................***Failed    2.07 sec
.F
======================================================================
FAIL: test_reload_dump (__main__.TestH5MD)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/espressopp/espressopp/testsuite/io/test_h5md_parallel.py", line 86, in test_reload_dump
    self.binary_compare('reference.h5', 'dump.h5')
  File "/__w/espressopp/espressopp/testsuite/io/test_h5md_parallel.py", line 35, in binary_compare
    self.assertEqual(b1, b2)
AssertionError: b'\x0[709 chars]00\x073Rb\x00\x00\x80\x00\x00\x00\x00\x00\x00\[1241 chars]\x00' != b'\x0[709 chars]00\x083Rb\x00\x00\x80\x00\x00\x00\x00\x00\x00\[1241 chars]\x00'

----------------------------------------------------------------------
Ran 2 tests in 0.051s

FAILED (failures=1)
Cubicity check passed -> powered by HeSpaDDA
HeSpaDDA message: Size Lenghts are eq. while ordering axis with preference on X, Y and Z!
nodeGrid:  (1, 1, 1)  cellGrid:  (7, 7, 7) half cell:  1
Cubicity check passed -> powered by HeSpaDDA
HeSpaDDA message: Size Lenghts are eq. while ordering axis with preference on X, Y and Z!
nodeGrid:  (1, 1, 1)  cellGrid:  (7, 7, 7) half cell:  1
```

**Comment by junghans** [comment]
This is part of the coverage validate workflow.

**Comment by XzzX** [comment]
https://github.com/espressopp/espressopp/runs/5958205136?check_suite_focus=true

**Comment by XzzX** [comment]
I cannot reproduce the error in my local docker container. Rerunning the job also fixed the problem here. We should keep an eye on it if this is a flaky test.

**Comment by junghans** [comment]
```
      Start 76: h5md_parallel
77/85 Test #76: h5md_parallel ........................***Exception: SegFault  1.28 sec
```
on `ubuntu_mpich`  not great, @XzzX can you have a look?

**Comment by XzzX** [comment]
looking into it.

**Comment by XzzX** [comment]
```
#58107 0x00007f6bf424e24f in H5_init_library () from /lib/x86_64-linux-gnu/libhdf5_mpich.so.103
#58108 0x00007f6bf424e80c in H5open () from /lib/x86_64-linux-gnu/libhdf5_mpich.so.103
#58109 0x00007f6be77b498f in espressopp::io::DumpH5MDParallel::dump (this=0x5589d3dc0e30)
```

remarkable, the stack is 58000 deep...

**Comment by junghans** [comment]
Not fixed in #435, @XzzX will try suggestions from https://github.com/espressopp/espressopp/pull/435#issuecomment-1178187881

**Comment by junghans** [comment]
Hopefully fixed by #438
